File: backend/health-platform/src/main/java/com/healthplatform/healthplatform/doctor/entity/Doctor.java
package com.healthplatform.healthplatform.doctor.entity;

import com.healthplatform.healthplatform.common.entity.BaseEntity;
import com.healthplatform.healthplatform.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "doctors")
@Getter
@Setter
@NoArgsConstructor
public class Doctor extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "user_id",
            nullable = false
    )
    private User user;

    @Column(
            name = "first_name",
            nullable = false,
            length = 100
    )
    private String firstName;

    @Column(
            name = "last_name",
            nullable = false,
            length = 100
    )
    private String lastName;

    @Column(
            name = "specialization",
            length = 150
    )
    private String specialization;

    @Column(
            name = "phone",
            length = 50
    )
    private String phone;

    @Column(
            name = "email",
            length = 255
    )
    private String email;

    @Column(
            name = "notes",
            columnDefinition = "TEXT"
    )
    private String notes;
}

File: frontend/src/types/doctor.ts
export type Doctor = {
  id: string;
  firstName: string;
  lastName: string;
  specialization: string | null;
  phone: string | null;
  email: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
};

export type CreateDoctorRequest = {
  firstName: string;
  lastName: string;
  specialization: string | null;
  phone: string | null;
  email: string | null;
  notes: string | null;
};

export type UpdateDoctorRequest = {
  firstName: string;
  lastName: string;
  specialization: string | null;
  phone: string | null;
  email: string | null;
  notes: string | null;
};
